### PR TITLE
Wfp 1325

### PR DESCRIPTION
**Pull request by carlov20**

### CODE REVIEW

**Review by markreesmoj — APPROVED** [review]
👍

File: helm_deploy/hmpps-probation-estate-api/values.yaml [hunk 1]
```diff
@@ -45,3 +45,4 @@ generic-service:
 
 generic-prometheus-alerts:
   targetApplication: hmpps-probation-estate-api
+  alertSeverity: hmpps-tier
```
Comment (sp-watson) [inline helm_deploy/hmpps-probation-estate-api/values.yaml:48]: Oh - is there where the alerts are going now? I thought it was `manage-a-workforce-dev`

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsprobationestateapi/integration/IntegrationTestBase.kt [hunk 2]
```diff
@@ -1,16 +1,64 @@
...
+import org.springframework.http.HttpHeaders
 import org.springframework.test.context.ActiveProfiles
 import org.springframework.test.web.reactive.server.WebTestClient
 
 @SpringBootTest(webEnvironment = RANDOM_PORT)
 @ActiveProfiles("test")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 abstract class IntegrationTestBase {
 
+  private var oauthMock: ClientAndServer =
```
Comment (sp-watson) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsprobationestateapi/integration/IntegrationTestBase.kt:25]: Interesting - I have not seen this `ClientAndServer` before - generally we use Wiremock directly

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -12,7 +12,11 @@ spring:
     date-format: "yyyy-MM-dd HH:mm:ss"
     serialization:
       WRITE_DATES_AS_TIMESTAMPS: false
-
+  security:
+    oauth2:
+      resourceserver:
+        jwt:
+          jwk-set-uri: ${oauth.endpoint.url}/.well-known/jwks.json
```
Comment (sp-watson) [inline src/main/resources/application.yml:19]: Actually that's much better than what we have done before - which was setting this url for each env

File: build.gradle.kts [hunk 4]
```diff
@@ -7,13 +7,26 @@ configurations {
...
+  implementation("org.springdoc:springdoc-openapi-data-rest:$springDocVersion")
+
+  testImplementation("io.jsonwebtoken:jjwt:0.9.1")
+  testImplementation("org.mock-server:mockserver-netty:5.13.2")
 }
 
 java {
   toolchain.languageVersion.set(JavaLanguageVersion.of(18))
 }
+repositories {
```
Comment (sp-watson) [inline build.gradle.kts:27]: Does this need to be specified explicitly - I thought it was the default?

File: helm_deploy/hmpps-probation-estate-api/values.yaml (hunk 1 above)
Comment (carlov20) [reply helm_deploy/hmpps-probation-estate-api/values.yaml:48]: `hmpps-tier` alert severity maps to the slack channel `manage_a_workforce_dev`, the name is a bit misleading

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsprobationestateapi/integration/IntegrationTestBase.kt (hunk 2 above)
Comment (carlov20) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsprobationestateapi/integration/IntegrationTestBase.kt:25]: I must admit I just followed suit with some of our other services which use mock server https://www.mock-server.com/ as the mocking mechanism

**Pull request merged**